lib/rubocop/cop/rspec/multiple_expectations.rb

```diff
@@ -77,10 +77,22 @@ def on_block(node)
         private
 
         def example_with_aggregated_failures?(node)
-          example = node.send_node
+          aggregated_failures?(node) ||
+            ancestors_with_aggregated_failures?(node)
+        end
+
+        def ancestors_with_aggregated_failures?(node)
+          node
+            .each_ancestor(:block)
+            .find { |ancestor| aggregated_failures?(ancestor) }
+        end
 
-          with_aggregated_failures?(example) &&
-            !disabled_aggregated_failures?(example)
+        def aggregated_failures?(node)
+          example = RuboCop::RSpec::Example.new(node)
+          example.metadata&.any? do |metadata|
```

Comment:
it would be so easier when we update to Rubocop 0.68 and start using [children matching](https://github.com/rubocop-hq/rubocop/pull/6965)

Reply:
Awesome, loving [it in action](https://github.com/rubocop-hq/rubocop/pull/6999)!

Reply:
Sent #773 to update RuboCop to 0.70.0+.
Going to update this pull request to use the new children matching syntax.

Reply:
For the following code:
```
code = <<-CODE
describe SomeClass, :inline, :aggregate_failures, :bulk do
  a
end
CODE
```

```
def match(pattern, source)
  ruby_version = RUBY_VERSION.to_f
  node_pattern = RuboCop::NodePattern.new(pattern)
  ast = RuboCop::ProcessedSource.new(source, ruby_version).ast
  node_pattern.match(ast)
end

match('(block (send $<(sym :aggregate_failures)>) ...)', code) # => nil
match('(block (send $<(sym :aggregate_failures)> ...) ...)', code) # => nil
match('(block (send $...) ...)', code)
# => [nil, :describe, s(:const, nil, :SomeClass), s(:sym, :inline), s(:sym, :aggregate_failures), s(:sym, :bulk)] 
```

Not sure how children matching can help me in this case. Can you please give an advise, maybe I'm missing something obvious?

Also, metadata can be in different positions, e.g. in `specify` without a docstring, only symbolic metadata, or only hash metadata, this is only getting more complicated.


Just a clarification why I'm trying to avoid using `def_node_search`:
```
describe SomeClass, :inline, aggregate_failures: false, :bulk do
  describe '#something' do
    # current example being examined
    it 'does something' do
      # ...
    end

    # sibling example
    it 'does something else', aggregate_failures: true do # I don't want it to match this
      # ...
    end
  end
end
```

When I do tree walking through parents, I only want to match parent's metadata, so doing a `def_node_search` might mistakenly find metadata from a sibling example. Or at least it's confusing.

Reply:
Sure, [added one](https://github.com/rubocop-hq/rubocop-rspec/pull/768/files#diff-7ce36e28810f5439e2331ed3b50431aaR180).